Framework/API/src/InfoComponentVisitor.cpp

```diff
@@ -137,5 +144,9 @@ size_t InfoComponentVisitor::size() const {
   return m_componentIds.size() - m_droppedDetectors;
 }
 
+const std::unordered_map<Mantid::Geometry::IComponent *, size_t> &
+InfoComponentVisitor::componentIdToIndexMap() const {
+  return m_componentIdToIndexMap;
```

Comment:
For extra speedup return by value here and use `return std::move(m_componentIdToIndexMap`, and the same for `componentIds()` above. That avoids two copies all in all.